odl/tomo/geometry/conebeam.py

```diff
@@ -0,0 +1,557 @@
+            The detector parameters
+        src_rad : `float`
+            Radius of the source circle, must be positive
+        det_rad : `float`
+            Radius of the detector circle, must be positive
+        agrid : 1-dim. `TensorGrid`, optional
+            A sampling grid for `angle_intvl`
+        dgrid : 2-dim. `TensorGrid`, optional
+            A sampling grid for `dparams`
+        angle_offset : `float`
+            Offset to the rotation angle in the azimuthal plane. Does not
+            affect the offset in z-direction.
+        axis : `int` or 3-element array
+            Defines the rotation axis via a 3-element vector or a single
+            integer referring to a standard axis
+        """
+        super().__init__(angle_intvl, src_rad, det_rad, agrid, angle_offset,
+                         axis)
+
+        if not (isinstance(dparams, IntervalProd) and dparams.ndim == 2):
+            raise TypeError('detector parameters {!r} are not an interval.'
+                            ''.format(dparams))
+
+        if dgrid is not None:
+            if not isinstance(dgrid, TensorGrid):
+                raise TypeError('detector grid {!r} is not a `TensorGrid` '
+                                'instance.'.format(dgrid))
+            if not dparams.contains_set(dgrid):
+                raise ValueError('detector grid {} not contained in detector '
+                                 'parameter interval {}.'
+                                 ''.format(dgrid, dparams))
+
+        self._detector = Flat2dDetector(dparams, dgrid)
+
+    @property
+    def detector(self):
+        """Detector of this geometry."""
+        return self._detector
+
+    def det_to_src(self, angle, dpar, normalized=True):
+        """Direction from a detector location to the source.
+
+        Parameters
+        ----------
+        angle : `float`
+            The motion parameter. It must be contained in this
+            geometry's motion parameter set
+        dpar : 2-tuple of `float`
+            The detector parameter. It must be contained in this
+            geometry's detector parameter set
+        normalized : `bool`, optional
+            If `False` return the vector from the detector point to the source
+            parametrized by ``dpar`` and ``angle``. If `True`, return the
+            normalized version of that vector. Default: `True`
+
+        Returns
+        -------
+        vec : `numpy.ndarray`, shape (`ndim`,)
+            (Unit) vector pointing from the detector to the source
+        """
+        if angle not in self.motion_params:
+            raise ValueError('angle {} is not in the valid range {}.'
+                             ''.format(angle, self.motion_params))
+        if dpar not in self.det_params:
+            raise ValueError('detector parameter {} is not in the valid '
+                             'range {}.'.format(dpar, self.det_params))
+
+        axis = self.axis
+        angle += self.angle_offset
```

Comment:
Result of a discussion, but now that I see it again, I admit it's not needed. One can now very easily just add an offset to the angle interval and use that in the initialization.